Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTipSpawner.cs

```diff
@@ -0,0 +1,289 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using System.Collections;
+using UnityEngine;
+using HoloToolkit.Unity.InputModule;
+
+namespace HoloToolkit.UX.ToolTips
+{
+    /// <summary>
+    /// Add to any Object to spawn ToolTips on tap or on focus, according to preference
+    /// Applies its follow settings to the spawned ToolTip's ToolTipConnector component
+    /// </summary>
+    public class ToolTipSpawner : MonoBehaviour , IInputHandler, IPointerSpecificFocusable
+    {
+        private enum VanishType
+        {
+            VanishOnFocusExit,
+            VanishOnTap,
+        }
+
+        private enum AppearType
+        {
+            AppearOnFocusEnter,
+            AppearOnTap,
+        }
+
+        public enum RemainType
+        {
+            Indefinite,
+            Timeout,
+        }
+
+
+        [SerializeField]
+        private Vector3 defaultDimensions = new Vector3(0.182f, 0.028f, 1.0f);
+
+        [SerializeField]
+        private bool showBackground = true;
+
+        [SerializeField]
+        private bool showOutline = false;
+
+        [SerializeField]
+        private bool showConnector = true;
+
+        [SerializeField]
+        private AppearType appearType = AppearType.AppearOnFocusEnter;
+
+        [SerializeField]
+        private VanishType vanishType = VanishType.VanishOnFocusExit;
+
+        [SerializeField]
+        private RemainType remainType = RemainType.Timeout;
+
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float appearDelay = 0.0f;
+
+        [SerializeField]
+        [Range(0f, 5f)]
+        private float vanishDelay = 2.0f;
+
+        [SerializeField]
+        [Range(0.5f, 10.0f)]
+        private float lifetime = 1.0f;
+
+        [SerializeField]
+        private GameObject toolTipPrefab;
+
+        [SerializeField]
+        private ConnectorFollowType followType = ConnectorFollowType.AnchorOnly;
+
+        [SerializeField]
+        private ConnnectorPivotMode pivotMode = ConnnectorPivotMode.Manual;
+
+        [SerializeField]
+        private ConnectorPivotDirection pivotDirection = ConnectorPivotDirection.North;
+
+        [SerializeField]
+        private ConnectorOrientType pivotDirectionOrient = ConnectorOrientType.OrientToObject;
+
+        [SerializeField]
+        private Vector3 manualPivotDirection = Vector3.up;
+
+        [SerializeField]
+        private Vector3 manualPivotLocalPosition = Vector3.up;
+
+        [SerializeField]
+        [Range(0f, 1f)]
+        private float pivotDistance = 0.25f;
+
+        [SerializeField]
+        private string ToolTipText = "New Tooltip";
+
+        [SerializeField]
+        private Transform Anchor;
+
```

Comment:
nit: extra blank line

Reply:
can you verify that you still see this extra line?